PizzaBuilder/PizzaBuilder/Program.cs

```diff
@@ -0,0 +1,182 @@
+        }
+
+        public void BuildTopping()
+        {
+            this.WhitePizza.Add("Topping");
+        }
+
+        public void reset()
+        {
+            this.WhitePizza = new whitePizza();
+        }
+
+        public whitePizza GetPizza()
+        {
+            whitePizza result = this.WhitePizza;
+            this.reset();
+            return result;
+        }
+    }
+    public class WheatPizzaBuilder : IPizzaBuilder
+    {
+        private wheatPizza wheatPizza = new wheatPizza();
+
+        public WheatPizzaBuilder()
+        {
+            this.reset();
+        }
+        public void BuildDough()
+        {
+            this.wheatPizza.Add("Wheat Dough");
+        }
+
+        public void BuildSauce()
+        {
+            this.wheatPizza.Add("Sauce");
+        }
+
+        public void BuildTopping()
+        {
+            this.wheatPizza.Add("Topping");
+        }
+
+        public void reset()
+        {
+            this.wheatPizza = new wheatPizza();
+        }
+
+        public wheatPizza GetPizza()
+        {
+            wheatPizza result = this.wheatPizza;
+            this.reset();
+            return result;
+        }
+    }
+
+    public class whitePizza
+    {
+        private List<object> ingredients = new List<object>();
+        public void Add(string ingredient)
+        {
+            this.ingredients.Add(ingredient);
+        }
+        public string listIngredients()
+        {
+            string str = "";
+            for (int i = 0; i < this.ingredients.Count; i++)
+            {
+                str += this.ingredients[i] + ",";
+            }
+            return "White Pizza ingredientsts:" + str;
+        }
+
+    }
+    public class wheatPizza
+    {
+        private List<object> ingredients = new List<object>();
+        public void Add(string ingredient)
+        {
+            this.ingredients.Add(ingredient);
+        }
+        public string listIngredients()
+        {
+            string str = "";
+            for (int i = 0; i < this.ingredients.Count; i++)
+            {
+                str += this.ingredients[i] + ",";
+            }
+            return "Wheat Pizza ingredientsts:" + str;
+        }
+
+    }
+
+
+    public class Director
+    {
+
+        private IPizzaBuilder _PizzaBuilder;
+        public IPizzaBuilder Builder
+        {
+            set { this._PizzaBuilder = value; }
+        }
+        public void makePizza()
+        {
+            this._PizzaBuilder.BuildDough();
+            this._PizzaBuilder.BuildSauce();
+            this._PizzaBuilder.BuildTopping();
+        }
+        public void makePizzaWithoutTopping()
+        {
+            this._PizzaBuilder.BuildDough();
+            this._PizzaBuilder.BuildSauce();
+       
+        }
+
+    }
+
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var director = new Director();
+            //builder White
+            var builderWhite = new WhitePizzaBuilder();
```

Comment:
It would be better if you used polymorphism to define builders, without creating new builder variable for each one.